WordPress/gutenberg
# Update block alignment setting to use "supports: align" block alignment hook instead of "align" attribute

## Description
This PR addresses #7098 and #7099 which reports a bug regarding the pull quote and gallery blocks. Upon un-assigning a block alignment setting from the blocks, it results in a "This block appears to have been modified externally" error due to a class of `alignundefined` to be added to the block against the expected `alignnone`.

I also discovered that button block also result in the same behavior so I've implemented changes in it in this PR. The categories, embed and table blocks too use the same old structure, so I've updated them as well.

## How has this been tested?
This has been tested following the steps below:

1. Add any one of the above mentioned blocks in the Gutenberg editor and assign a block alignment setting (e.g. Right) to it.
2. Publish/Update your post.
3. Un-assign the block alignment setting done in Step 1.
4. Update the post and reload the page, see that the reported error message doesn't appear.

This was tested in WP 4.9.6, Gutenberg 2.9.2, Apache server with PHP 7.2.0 and MySQL 5.6.34. According to initial tests, the code doesn’t seem to affect any other areas.

## Screenshots 
![pull-7100](https://user-images.githubusercontent.com/20284937/40872558-261c7f44-6672-11e8-886f-4ad06a83c68d.gif)

## Types of changes
This PR completely strips off the `align` attribute from the blocks and implements the newly introduced `supports: align` block alignment hook.

## Note
Validation seems to have failed as I've not updated the unit tests. To any kind reviewer who's reviewing this PR, please let me know if the changes look good and if I should work on the unit tests.

## Checklist:
